tutorial/SparkRedshiftTutorial.scala

```diff
@@ -0,0 +1,172 @@
+		val awsAccessKey = args(0)
+		val awsSecretKey = args(1)
+		val rsDbName = args(2)
+		val rsUser = args(3)
+		val rsPassword = args(4)
+    /** Sample Redshift URL is swredshift.czac2vcs84ci.us-east-1.redshift.amazonaws.com:5439 */
+		val rsURL = args(5)  
+		val jdbcURL = s"""jdbc:redshift://$rsURL/$rsDbName?user=$rsUser&password=$rsPassword"""
+    println(jdbcURL)
+		val sc = new SparkContext(new SparkConf().setAppName("SparkSQL").setMaster("local"))
+
+		val tempS3Dir = "s3n://redshift-spark/temp/"
+		sc.hadoopConfiguration.set("fs.s3n.awsAccessKeyId", awsAccessKey)
+		sc.hadoopConfiguration.set("fs.s3n.awsSecretAccessKey", awsSecretKey)
+
+		val sqlContext = new org.apache.spark.sql.SQLContext(sc)
+
+		import sqlContext.implicits._
+    /** Load from a table */
+		val eventsDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir) 
+			.option("dbtable", "event") 
+			.load()
+		eventsDF.show()
+		eventsDF.printSchema()
+
+    /** Load from a query */
+		val salesQuery = """select salesid,listid,sellerid,buyerid,
+                        eventid,dateid,qtysold,pricepaid,commission 
+                        from sales 
+                        order by saletime desc LIMIT 10000"""
+		val salesDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL)
+			.option("tempdir", tempS3Dir)
+			.option("query", salesQuery)
+			.load()
+		salesDF.show()
+
+		val eventQuery = "select * from event"
+		val eventDF = sqlContext.read
+			.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir)
+			.option("query", eventQuery)
+			.load()
+
+    /** Register 'event' table as 'myevent' in the Spark Environment */
+		eventsDF.registerTempTable("myevent") 
+
+    /** Save to a redshift table from a table registered in spark */
+    
+    /*
+     * Create a new table redshiftevent after dropping any existing redshiftevent table
+     * and write event records with event id less than 1000
+     */
+		sqlContext.sql("select * from myevent where eventid<=1000").withColumnRenamed("eventid", "id")
+			.write.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL)
+			.option("tempdir", tempS3Dir)
+			.option("dbtable", "redshiftevent")
+			.mode(SaveMode.Overwrite)
+			.save()
+    
+    /*
+     * Append to an existing table redshiftevent if it exists or create a new one if it does not
+     * exist and write event records with event id greater than 1000
+     */
+		sqlContext.sql("select * from myevent where eventid>1000").withColumnRenamed("eventid", "id")
+			.write.format("com.databricks.spark.redshift")
+			.option("url", jdbcURL) 
+			.option("tempdir", tempS3Dir)
+			.option("dbtable", "redshiftevent")
+			.mode(SaveMode.Append) 
+			.save()
+
+		/** Demonstration of Inter-operability */
+		val salesAGGQuery = """select sales.eventid as id,sum(qtysold) as totalqty,sum(pricepaid)  as salesamt
```

Comment:
Spacing is inconsistent here (two spaces before the `as` instead of one). These details matter for public-facing example code.